### PR TITLE
load sensors_3d.yaml

### PR DESCRIPTION
**Pull request by mikeferguson**
### Description

this implements #1297 - also requires some changes to the MSA, which will be a separate PR to that feature branch. The changes are namely:

 * "sensors" is expected to be a vector of strings (sensor names)
 * the sensor config then needs to be namespaced under the sensor name provided
 * all of the distance/padding parameters need to be floating point - not integer or string.
 
 Example of changes to get a working configuration: https://github.com/mikeferguson/ubr_reloaded/commit/45b9e52ac6ef04c9a0fd94f8a2d18956739a278f

Screenshot from 2022-06-22 23-07-45

Please explain the changes you made, including a reference to the related issue if applicable

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by vatanaksoytezer** [comment]
@mikeferguson thank you so much for the PR! I'm unsure why the CI is failing, the failure seems unrelated to this PR, I'll look into it. Pasting the error message just for reference:

```
2022-06-23T03:31:36.4720982Z 3: Test command: /usr/bin/python3.10 "-u" "/opt/ros/humble/share/ament_cmake_test/cmake/run_test.py" "/home/runner/work/moveit2/moveit2/.work/target_ws/build/moveit_kinematics/test_results/moveit_kinematics/test_launch_fanuc-kdl-singular.test.py.xunit.xml" "--package-name" "moveit_kinematics" "--output-file" "/home/runner/work/moveit2/moveit2/.work/target_ws/build/moveit_kinematics/ros_test/test_launch_fanuc-kdl-singular.test.py.txt" "--command" "ros2" "test" "/home/runner/work/moveit2/moveit2/.work/target_ws/src/moveit2/moveit_kinematics/test/launch/fanuc-kdl-singular.test.py" "test_binary_dir:=/home/runner/work/moveit2/moveit2/.work/target_ws/build/moveit_kinematics/test" "--junit-xml=/home/runner/work/moveit2/moveit2/.work/target_ws/build/moveit_kinematics/test_results/moveit_kinematics/test_launch_fanuc-kdl-singular.test.py.xunit.xml" "--package-name=moveit_kinematics"
2022-06-23T03:31:36.4722302Z 3: Test timeout computed to be: 60
2022-06-23T03:31:36.4722835Z 3: -- run_test.py: invoking following command in '/home/runner/work/moveit2/moveit2/.work/target_ws/build/moveit_kinematics/test':
2022-06-23T03:31:36.4724266Z 3:  - ros2 test /home/runner/work/moveit2/moveit2/.work/target_ws/src/moveit2/moveit_kinematics/test/launch/fanuc-kdl-singular.test.py test_binary_dir:=/home/runner/work/moveit2/moveit2/.work/target_ws/build/moveit_kinematics/test --junit-xml=/home/runner/work/moveit2/moveit2/.work/target_ws/build/moveit_kinematics/test_results/moveit_kinematics/test_launch_fanuc-kdl-singular.test.py.xunit.xml --package-name=moveit_kinematics
2022-06-23T03:31:36.4725027Z 3: Running with ROS_DOMAIN_ID 1
2022-06-23T03:31:36.4725266Z 3: ROS_DOMAIN_ID 1
2022-06-23T03:31:36.4725498Z 3: Failed to handle type {}
```

**Comment by mikeferguson** [comment]
@vatanaksoytezer don't worry about the build failure quite yet - at least part of it is this PR - the various robot configs in moveit_resources are all missing the sensors_3d.yaml (so the first time around it failed due to missing file - I added a file.exists - but now I see that causes a parsing issue later since it ends up putting an empty dictionary into some of the tests - I'll work on resolving that today)

File: moveit_configs_utils/moveit_configs_utils/moveit_configs_builder.py
Comment (JafarAbdi) [inline moveit_configs_utils/moveit_configs_utils/moveit_configs_builder.py:125]: If I understand this correctly, it was failing because `sensors_3d` is None and calling `update` with None is throwing an exception?

Comment (mikeferguson) [reply moveit_configs_utils/moveit_configs_utils/moveit_configs_builder.py:125]: Based on the error message - I think it was actually that it added an empty dictionary inside the list of parameters (although I may be misreading the somewhat cryptic error)

Comment (mikeferguson) [reply moveit_configs_utils/moveit_configs_utils/moveit_configs_builder.py:125]: Ah - ok - the sensors_3d.yaml file EXISTS, but is empty..

File: moveit_configs_utils/moveit_configs_utils/moveit_configs_builder.py
Comment (mikeferguson) [inline moveit_configs_utils/moveit_configs_utils/moveit_configs_builder.py:356]: the second part of this (that the first sensor is non-empty) can be removed once https://github.com/ros-planning/moveit_resources/pull/141 is released

**Comment by mikeferguson** [comment]
@vatanaksoytezer @DLu this is now ready for a review (tests are passing, I'd suggest we merge with the extra logic and I'll come back and remove that extra checking in a month or so once the moveit_resources stuff percolates through the build farm)

**Comment by vatanaksoytezer** [comment]
I just released moveit_resources for rolling and humble with David's fixes: https://github.com/ros/rosdistro/pull/33703 and https://github.com/ros/rosdistro/pull/33702. So we are just waiting them to get merged and synced. @mikeferguson the PR looks good to me, I'm happy to approve and merge if you don't mind adding a TODO and open up an issue to remove it so we don't forget. Thank you very much!

**Comment by mikeferguson** [comment]
@vatanaksoytezer TODO has been added (it should only take a few weeks for a sync to happen and then we can remove the code). This has been rebased after the massive feature/msa merge - if you have time to do a review would like to get this in.

**Review by vatanaksoytezer — APPROVED** [review]

**Pull request merged**
